src/awkward/operations/ak_to_torch.py

```diff
@@ -0,0 +1,67 @@
+# BSD 3-Clause License; see https://github.com/scikit-hep/awkward/blob/main/LICENSE
+
+from __future__ import annotations
+
+import awkward as ak
+from awkward._dispatch import high_level_function
+from awkward._nplikes.numpy_like import NumpyMetadata
+
+__all__ = ("to_torch",)
+
+np = NumpyMetadata.instance()
+
+
+@high_level_function()
+def to_torch(array):
+    """
+    Args:
+        array: Array-like data. May be a high level #ak.Array,
+            or low-level #ak.contents.ListOffsetArray, #ak.contents.ListArray,
+            #ak.contents.RegularArray, #ak.contents.NumpyArray
+
+    Converts `array` (only ListOffsetArray, ListArray, RegularArray and NumpyArray data types supported)
+    into a PyTorch Tensor, if possible.
+
+    If `array` contains any other data types (RecordArray for example) the function raises a TypeError.
+    """
+
+    # Dispatch
+    yield (array,)
+
+    # Implementation
+    return _impl(array)
+
+
+def _impl(array):
+    try:
+        import torch
+    except ImportError as err:
+        raise ImportError(
+            """to use ak.to_torch, you must install 'torch' package with:
+
+        pip install torch or conda install pytorch"""
+        ) from err
+
+    # useful function that handles all possible input arrays
+    array = ak.to_layout(array, allow_record=False)
+
+    # get the device array is on
+    device = ak.backend(array)
+
+    # convert to numpy or cupy if `array` on gpu
+    try:
+        np_array = array.to_backend_array(allow_missing=False)
```

Comment:
`np_array` has a misleading name (but I can't "suggest change" for that because it affects multiple lines).

Also, you might not have known about it, but there are more backend types than `"cpu"` and `"cuda"`. I think it would be reasonable for this function to raise an error if the `array` is not one of these two. (So maybe there is a reason to get the `device`, after all!) The other _currently existing_ backends are `"typetracer"` for Dask-delayed data and `"jax"` for autodiff (which might be replaced). `ak.to_torch` is an inherently eager function, so it can't be used with Dask and raising an error would be a reasonable thing to do.